src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -75,16 +75,28 @@ default FilePath getFilePathForUniqueFolder(Launcher launcher, String uniqueName
         /*Use of Computer is not recommended as jenkins hygeine for pipeline support
          * https://javadoc.jenkins-ci.org/jenkins/tasks/SimpleBuildStep.html */
         
-        String tmpDir = getNodeSpecificTmpFolderPath(workspace);
-        return new FilePath(launcher.getChannel(), tmpDir+"/"+uniqueName);
+        String tmpDir = getNodeSpecificTmpFolderPath(workspace, launcher);
+        String fileSeperator = "/";
+
+        if (!launcher.isUnix()){
+            fileSeperator = "\\";
+        }
+
+        return new FilePath(launcher.getChannel(), tmpDir + fileSeperator + uniqueName);
     }
 
-    default String getNodeSpecificTmpFolderPath(FilePath workspace) throws IOException, InterruptedException {
+    default String getNodeSpecificTmpFolderPath(FilePath workspace, Launcher launcher) throws IOException, InterruptedException {
         Computer cmp = workspace.toComputer();
         if (cmp == null) {
             throw new IOException(Message.getValue("build.workspace.computer.not.found"));
         }
         String tmpDir = (String) cmp.getSystemProperties().get("java.io.tmpdir");
+
+        // Handle Java temp folder path discrepancy for Windows.
+        if (!launcher.isUnix() && tmpDir.charAt(tmpDir.length() - 1) == '\\'){

```

Comment:
We are already verifying the launcher platform  under getProcessToRunMatlabCommand() method .  Can we bring this to that code instead passing Launcher as an extra parameter? Or else even better way is to handdle it at this line matlabLauncher.cmds(tmpDir + "\\" + runnerScriptName, "\"" + matlabCommand + "\"") if tmpdir includes "\" by default.